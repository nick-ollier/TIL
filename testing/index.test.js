import {
    getHex,
    HeaderComponent,
    getFlyingSuperheroes,
    HeaderWrapperComponent,
    DebugComponent
} from "./index";
import React from "react";
import { render } from "@testing-library/react";

// getHex

test("returns #ff0000", () => {
    expect(getHex("red")).toBe("#ff0000");
});

test("returns #000000 as a fallback", () => {
    expect(getHex("not actually a colour")).toBe("#000000");
});

// HeaderComponent

test("component renders", () => {
    render(<HeaderComponent />);
});

test("component still renders", () => {
    // "container" is always a div
    const { container } = render(<HeaderComponent />);
    expect(container.firstChild);
    console.log(container.innerHTML);
});

// getFlyingSuperheroes

test("returns array of superheroes that can fly - manual", () => {
    const flyingSuperheroes = getFlyingSuperheroes();
    expect(flyingSuperheroes).toEqual([
        { name: "Dynaguy", powers: ["disintegration ray", "fly"] },
        { name: "Apogee", powers: ["gravity control", "fly"] }
    ]);
});

test("returns array of superheroes that can fly - snapshot", () => {
    const flyingSuperheroes = getFlyingSuperheroes();
    console.log(flyingSuperheroes);
    expect(flyingSuperheroes).toMatchSnapshot();
});

// HeaderWrapperComponent

test("component matches snapshot", () => {
    const { container } = render(
        <HeaderWrapperComponent header="New header" />
    );
    expect(container.firstChild).toMatchSnapshot();
});

test("**component debug**", () => {
    const { container, debug } = render(
        <HeaderWrapperComponent header="New header" />
    );
    debug(container.firstChild);
});

// DebugComponent
// chrome://inspect -or- node debug button in dev menu
// debugger; in component
// not working right now - module 4 video 12
test("**component debug in browser**", () => {
    const { container, debug } = render(<DebugComponent />);
    debug(container);
});
