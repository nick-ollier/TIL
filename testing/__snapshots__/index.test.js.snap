// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches snapshot 1`] = `
<header>
  <h1>
    New header
  </h1>
</header>
`;

exports[`returns array of superheroes that can fly - snapshot 1`] = `
Array [
  Object {
    "name": "Dynaguy",
    "powers": Array [
      "disintegration ray",
      "fly",
    ],
  },
  Object {
    "name": "Apogee",
    "powers": Array [
      "gravity control",
      "fly",
    ],
  },
]
`;
